### ISSUE TITLE
CAT is activated but AirDrop and Handoff  do not work [Solved]

### ISSUE THREAD

**Issue by xyliutw** [dokterdok/Continuity-Activation-Tool#412]
Have you tried [these steps](https://github.com/dokterdok/Continuity-Activation-Tool/wiki#cant-open-the-tool)?
yes
Is there already an open topic for this issue?
not that I found
Please include the following information.

- OS Version: macOS High Sierra 10.13.1
- Mac model:  Macbook Pro(i7, 15-inch, Early 2011)
- Dongle or replaced card?:   Dongle
- Used dongle/card: BCM20702A0

Your Issue:
I disabled System Integrity Protection using Recovery and terminal "csrutil disable" and  install the Continuity Activation Tool. The System Diagnostic shows below
![image](https://user-images.githubusercontent.com/33772837/32977302-8510af0e-cc65-11e7-9d73-e3042848384f.png)

The view of AirDrop window has changed but I cannot detect any devices on it. Also, the handoff function in System Appearance -> General has enabled but still doesn't work.

![image](https://user-images.githubusercontent.com/33772837/32977327-379ce73c-cc66-11e7-93d1-36e0f8634448.png)

![image](https://user-images.githubusercontent.com/33772837/32977328-3d3763e8-cc66-11e7-86e2-3f50a9dfaf9c.png)

![image](https://user-images.githubusercontent.com/33772837/32977212-c6dccc58-cc63-11e7-821f-d874724c20b2.png)

Should I go to buy 4CAX card or there are any solutions on it? I only need to use AirDrop but it cannot work correctly ):

**Comment by kude87** [comment]
Hi, how did you fix this?  I am having the same problem.

**Comment by GGitR** [comment]
Yes please. How did you fix this? I am also having the same problem.

**Comment by kude87** [comment]
My iMessage is also flakey from my mac.  It only shows 5 conversations, and all are iMessage.  Do regular sms to android users not show up on the mac?  My calls do work on my mac, but I was really wanting iMessage and Airdrop the most.  Thoughts?

**Comment by GGitR** [comment]
AirDrop and iMessage work fine for me, even with SMS to non-iPhones; not sure it has anything to do with Hand Off though; that’s an iCloud thing.

**Comment by kude87** [comment]
Weird.  I've tried all the steps apple suggests for getting both to work, to no avail.  Not sure what else to do.  I've been a long time android user finally wanting to see how this all works together, and it's been less than impressive.  I'm still troubleshooting like I was on android.

**Comment by xyliutw** [comment]
Hello, I fixed it using  Hardware IO Tools for Xcode from the Apple Developers site (just search for it). Then, run Bluetooth Explorer. Choose ‘HCI Controller Selector’ from the tools menu. Actually, the dongle is worked after CAT performed. The reason why the AirDrops doesn't work is due to the default HCI selector. Just followed the instruction from the site below and then you can using AirDrops successfully.
(https://justhackme.wordpress.com/2014/05/21/adding-a-ble-bluetooth-4-0-dongle-to-mac-os-x-mavericks/)

**Comment by sparky672** [comment]
>  I fixed it using Hardware IO Tools for Xcode from the Apple Developers site (just search for it)

@xyliutw, what version are you using?   The IO Tools package you mention only goes up to Xcode version 7.3 - "Hardware IO Tools for Xcode 7.3".  However, I'm running Sierra and my Xcode is version 9.2

FWIW, be aware that your recommendations may be out of date for the latest OS.

**Comment by xyliutw** [comment]
@sparky672 The OS version I executed the tool is in 10.13.1 High Sierra, and the Xcode version is 9.1.
I noticed that the Hardware IO Tools for Xcode "7.3" but I executed it successfully at that time. Sorry that I did not mention the version of my Xcode version. Thanks for your reminder.
